### PR TITLE
Better handling of killed applications

### PR DESCRIPTION
**Pull request by hungj**
- Fix issue where HDFS resources not deleted when application killed
- Killing YARN application should fail client immediately
- Re-sort logs printed in client

### CODE REVIEW

File: tony-core/src/main/java/com/linkedin/tony/TonyClient.java [hunk 1]
```diff
@@ -686,12 +678,17 @@ private void cleanUp() {
...
       }
-      FileSystem fs = FileSystem.get(hdfsConf);
-      if (appResourcesPath != null && fs.exists(appResourcesPath)) {
-        fs.delete(appResourcesPath, true);
-      }
     } catch (IOException | YarnException e) {
-      LOG.error("Failed to clean up temporary files :" + appResourcesPath, e);
+      LOG.error("Failed to finish application.", e);
+    } finally {
+      try {
```
Comment (oliverhu) [inline tony-core/src/main/java/com/linkedin/tony/TonyClient.java]: can we create a util method for this `try-catch` blob? This looks the same as the one above

Comment (hungj) [reply tony-core/src/main/java/com/linkedin/tony/TonyClient.java]: hmm...how do we refactor a try-catch block? we'd have to pass the block as an argument to the util function.

**Comment by hungj** [comment]
hmm, I added a slightly different test case, if the amRpcClient is nulled out after run then it should no longer be possible to communicate with AM.

File: tony-core/src/main/java/com/linkedin/tony/TonyClient.java (hunk 1 above)
Comment (oliverhu) [reply tony-core/src/main/java/com/linkedin/tony/TonyClient.java]: i mean having a 
```
deleteFile(path, msg) {
   try {
   } catch {
      Log.error(msg)
   }
}
```

Comment (hungj) [reply tony-core/src/main/java/com/linkedin/tony/TonyClient.java]: Ah gotcha...fixed

**Pull request merged**